tests/gcp/hooks/test_bigquery.py

```diff
@@ -407,6 +407,62 @@ def test_validate_src_fmt_configs(self):
         assert "compatibility_val" in src_fmt_configs, \
             "_validate_src_fmt_configs should add backward_compatibility config"
 
+    def test_run_load_with_avro_as_src_fmt(self):
+        with mock.patch.object(hook.BigQueryBaseCursor, 'run_with_configuration'):
+            bq_hook = hook.BigQueryBaseCursor(mock.Mock(), 12345)
+
+            try:
+                bq_hook.run_load(
+                    destination_project_dataset_table='my_dataset.my_table',
+                    source_uris=[],
+                    source_format='AVRO',
+                    autodetect=True
+                )
+            except ValueError:
+                self.fail("run_load() raised ValueError unexpectedly!")
+
+    def test_run_load_with_parquet_as_src_fmt(self):
+        with mock.patch.object(hook.BigQueryBaseCursor, 'run_with_configuration'):
+            bq_hook = hook.BigQueryBaseCursor(mock.Mock(), 12345)
+
+            try:
+                bq_hook.run_load(
+                    destination_project_dataset_table='my_dataset.my_table',
+                    source_uris=[],
+                    source_format='PARQUET',
+                    autodetect=True
+                )
+            except ValueError:
+                self.fail("run_load() raised ValueError unexpectedly!")
+
+    def test_run_load_with_json_as_src_fmt(self):
+        with mock.patch.object(hook.BigQueryBaseCursor, 'run_with_configuration'):
+            bq_hook = hook.BigQueryBaseCursor(mock.Mock(), 12345)
+
+            try:
+                bq_hook.run_load(
+                    destination_project_dataset_table='my_dataset.my_table',
+                    source_uris=[],
+                    source_format='NEWLINE_DELIMITED_JSON',
+                    autodetect=True
+                )
+            except ValueError:
+                self.fail("run_load() raised ValueError unexpectedly!")
+
+    def test_run_load_with_datastore_backup_as_src_fmt(self):
+        with mock.patch.object(hook.BigQueryBaseCursor, 'run_with_configuration'):
+            bq_hook = hook.BigQueryBaseCursor(mock.Mock(), 12345)
+
+            try:
+                bq_hook.run_load(
+                    destination_project_dataset_table='my_dataset.my_table',
+                    source_uris=[],
+                    source_format='DATASTORE_BACKUP',
+                    autodetect=True
+                )
+            except ValueError:
+                self.fail("run_load() raised ValueError unexpectedly!")
+
```

Comment:
WDYT about using `parameterized` library to iterate over the tested parameters? It will eliminate redundancy since the tests are differ only slightly.
Here is some example:
https://github.com/apache/airflow/blob/master/tests/gcp/operators/test_cloud_build.py#L66
(be aware that running single parameterized test may be problematic but running the whole test class should be fine, also worth to mention is that if parameterized test raise Exception the traceback is little bit confusing)

Reply:
WDYT @mik-laj ?